### DISCUSSION TITLE
Add `schema.json` to the `turbo` package

### DISCUSSION THREAD

**Discussion by scottwillmoore** [vercel/turborepo#7466] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

I noticed that [Rome](https://rome.tools/) doesn't use a traditional hosted JSON schema and instead they publish their JSON schema in their package so that it can then be referenced with a local path:

```json
{
  "$schema": "./node_modules/rome/configuration_schema.json",
}
```

I see three advantages to this approach...

1. The JSON schema will match the installed version of the package.
2. The JSON schema will work without internet access, provided that `node_modules` has been installed.
3. Any changes to the JSON schema in the development can be used straight away in examples, etc.

### Describe the solution you'd like

Consider publishing the JSON schema for the `turbo.json` configuration with the `turbo` package. You could still leave the hosted JSON schema (which resolves to the latest version of `turbo as an option, but also provide a local JSON schema. I'm happy to work on a PR 😇.

### Describe alternatives you've considered

Continue to use the hosted JSON schema.

**Comment by chris-olszewski** [comment]
This seems reasonable to me, I'll try to get this in sometime today.

**Comment by anthonyshew** [comment]
This is interesting but does have one small drawback: You'd have to install packages for the node_modules reference to work. I'm not sure if there are situations where that would create more pain than referencing the schema from the web.

(I don't see this comment turning out to be a steelman against doing this - but do want to make sure we think through what gotchas may exist.)

**Reply by anthonyshew** [reply]
Alternatively, why not do both?
